crates/bdk/src/wallet/mod.rs

```diff
@@ -443,6 +443,20 @@ impl<D> Wallet<D> {
             .map(|((k, i), full_txo)| new_local_utxo(k, i, full_txo))
     }
 
+    /// List all relevant outputs (includes both spent and unspent).
```

Comment:
Does this include both confirmed and unconfirmed outputs? In any case, can this be specified?

Also, we should specify that floating outputs are not included, as said in the `try_filter_chain_txouts` docs.

Reply:
> Does this include both confirmed and unconfirmed outputs? In any case, can this be specified?

Yes it does. I'll mention it.

> Also, we should specify that floating outputs are not included, as said in the try_filter_chain_txouts docs.

Do we want to expose the concept of "floating outputs" to the `Wallet`? The documentation of `Wallet::insert_txout` already states; *"This can be used to add a `TxOut` that the wallet does not own..."*.

Reply:
> Do we want to expose the concept of "floating outputs" to the Wallet?

I agree that we might not want that, but then list_outputs should also list floating outputs...

Reply:
@danielabrozzoni what I mean is that the exposed methods to add a "floating `TxOut`" explicitly state that the output will not be added as an owned output (we can make these docs clearer).

Non-owned outputs won't be listed.

Reply:
As discussed, we should improve docs of `insert_txout`!

Reply:
Clarified docs in 278210bb8918923005e80968df3dfcc3372c0a5c